### PR TITLE
Improves flag formatting for kerberos ticket presenter

### PR DESCRIPTION
**Pull request by cgranleese-r7**
This PR adds support for flag formatting to the Kerberos ticket presenter.

Expand this output to now breakdown what flags are enabled with a breif description of each flag:
```

  Group IDs:
      Relative ID: 513
      Attributes: 7
        ..0. .... .... .... .... .... .... .... Resource: The RESOURCE bit is NOT SET
       .... .... .... .... .... .... .... 0... Owner: The OWNER bit is NOT SET
       .... .... .... .... .... .... .... .1.. Enabled: The ENABLED bit is SET
        .... .... .... .... .... .... .... ..1. Enabled By Default: The ENABLED_BY_DEFAULT bit is SET
        .... .... .... .... .... .... .... ...1 Mandatory: The MANDATORY bit is SET
      Relative ID: 512
      Attributes: 7
        ..0. .... .... .... .... .... .... .... Resource: The RESOURCE bit is NOT SET
        .... .... .... .... .... .... .... 0... Owner: The OWNER bit is NOT SET
        .... .... .... .... .... .... .... .1.. Enabled: The ENABLED bit is SET
        .... .... .... .... .... .... .... ..1. Enabled By Default: The ENABLED_BY_DEFAULT bit is SET
        .... .... .... .... .... .... .... ...1 Mandatory: The MANDATORY bit is SET
```

## Before
image

## After
image

## Verification

- [ ] Create a `.kirbi` file to test against - https://github.com/rapid7/metasploit-framework/pull/17635
- [ ] Start `msfconsole`
- [ ] Run `run ticket_path=./ticket.kirbi NTHASH=<NTHASH>`

### CODE REVIEW

File: spec/lib/rex/proto/kerberos/credential_cache/krb5_ccache_presenter_spec.rb [hunk 1]
```diff
@@ -293,5 +334,64 @@
         EOF
       end
     end
+
+    describe '#print_bindata_model' do
+
+      context 'when the bindata object has fields that are not 1bit' do
+        it 'should raise an exception' do
+          binary_value = [7].pack('C')
```
Comment (adfoster-r7) [inline spec/lib/rex/proto/kerberos/credential_cache/krb5_ccache_presenter_spec.rb:342]:
```suggestion
          binary_value = [7].pack('N')
```

File: spec/lib/rex/proto/kerberos/credential_cache/krb5_ccache_presenter_spec.rb [hunk 2]
```diff
@@ -293,5 +334,64 @@
...
+          group_attributes = Rex::Proto::Kerberos::Pac::GroupAttributes.read(binary_value)
+          expect { subject.print_bindata_model(group_attributes, bit_length: 2) }.to raise_error RuntimeError, 'Bit length(2) should not be less that the number of flags(32) within the BinData model'
+        end
+      end
+
+      context 'with binary value' do
+        it 'should return formatted flag descriptions for GROUP_MEMBERSHIP flags with RESOURCE set' do
+          binary_value = [536870912].pack('N')
+          group_attributes = Rex::Proto::Kerberos::Pac::GroupAttributes.read(binary_value)
+          expect(subject.print_bindata_model(group_attributes, bit_length: 32)).to eq <<-EOF.rstrip
```
Comment (adfoster-r7) [inline spec/lib/rex/proto/kerberos/credential_cache/krb5_ccache_presenter_spec.rb:360]: should be able to update the indentation on the left if you use `to match_table <<~TABLE` instead

**Comment by cgranleese-r7** [comment]
@msjenkins-r7 retest this please

File: lib/rex/proto/kerberos/pac/krb5_pac.rb [hunk 3]
```diff
@@ -39,6 +39,103 @@ class Krb5SidAndAttributes < RubySMB::Dcerpc::Ndr::NdrStruct
...
+
+    bit1 :used_ntlmv2_auth_and_session_key
+    bit1 :profile_path_populated
+    bit1 :resource_group_ids
+    bit1 :accepts_ntlmv2
+
+    bit1 :machine_account
+    bit1 :sub_authentication
+    bit1 :extra_sids
+    bit1 :_reserved_27
```
Comment (adfoster-r7) [inline lib/rex/proto/kerberos/pac/krb5_pac.rb:81]:
```suggestion
    1.times do
      bit1 :"_reserved_#{self.fields.length}"
    end
```

File: lib/rex/proto/kerberos/pac/krb5_pac.rb [hunk 4]
```diff
@@ -39,6 +39,103 @@ class Krb5SidAndAttributes < RubySMB::Dcerpc::Ndr::NdrStruct
...
+
+    bit1 :lan_manager
+    bit1 :_reserved_29
```
Comment (adfoster-r7) [inline lib/rex/proto/kerberos/pac/krb5_pac.rb:84]:
```suggestion
    1.times do
      bit1 :"_reserved_#{self.fields.length}"
    end
```

**Comment by adfoster-r7** [comment]
Working as expected

**Comment by adfoster-r7** [comment]
## Release Notes

The `auxiliary/admin/kerberos/inspect_ticket` and `auxiliary/admin/kerberos/forge_ticket` modules have been updated to visually represent the decoded binary values of the Kerberos ticket fields

**Pull request merged**